pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -397,9 +401,13 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 		var domains []string
 		var altHosts []string
 		if svc == nil {
-			domains = []string{util.IPv6Compliant(hostname), name}
+			if SupportsIgnorePort(node) {
```

Comment:
why need to ignore port in sidecar?  Shouldn't we only concentrate on the gateway?

Reply:
I agree gateway is more important but it seems like we shouldn't ever be looking at the port?

Reply:
I remember the original issue is on gateway, the access from remote users can use a port totally different from the gateway service because of lb in front.

But for sidecar, we do not meet such problem. We should not change the behavior, in case we miss some risks

Reply:
Ok, I added a feature flag for sidecar